konduit-serving-core/src/main/java/ai/konduit/serving/configprovider/PipelineRouteDefiner.java

```diff
@@ -212,6 +224,74 @@ public Router defineRoutes(Vertx vertx, InferenceConfiguration inferenceConfigur
             }
         });
 
+        /**
+         * Sets up and endpoint to see server logs if
+         * {@link ServingConfig#isCreateLoggingEndpoints()} is true.
+         * Returns the last 100 lines of the log file.
+         */
+        if(inferenceConfiguration.getServingConfig().isCreateLoggingEndpoints()) {
+            router.get("/logs")
+                    .handler(ctx -> {
+                try {
+                    ctx.response()
+                            .putHeader(HttpHeaders.CONTENT_TYPE, "text/html")
+                            .end(FileUtils.readFileToString(new ClassPathResource("web/logs_page.html").getFile(), Charset.defaultCharset()));
+                } catch (Exception e) {
+                    ctx.fail(500, e);
+                }
+            });
+
+        /**
+         * Sets up and endpoint to see server logs if {@link ServingConfig#isCreateLoggingEndpoints()}
+         * is true. Returns the number of last few lines determines by the path param
+         * {@code numberOfLastLinesToRead}. If the path param is an invalid integer,
+         * then the whole log file data will be read and returned.
+         */
+            router.get("/logs/:numberOfLastLinesToRead")
+                    .handler(ctx -> {
+                String numberOfLinesString = ctx.pathParam("numberOfLastLinesToRead");
+
+                try {
+                    ctx.response().putHeader(HttpHeaders.CONTENT_TYPE, "text/plain").end(LogUtils.getLogs(Integer.parseInt(numberOfLinesString)));
+                } catch (Exception e) {
+                    if(e instanceof NumberFormatException) {
+                        if("download".equalsIgnoreCase(numberOfLinesString)) {
+                            ctx.response().sendFile(LogUtils.getLogsFile().getAbsolutePath()).end();
+                        } else if("downloadAsZip".equalsIgnoreCase(numberOfLinesString)) {
```

Comment:
Let's clean this up a bit... don't try to parse integer and catch exceptions
Instead, something like:
```
response = ctx.response().putHeader(HttpHeaders.CONTENT_TYPE, "text/plain");
if(numberOfLinesString.matches("\\d+"){ //Or maybe "-?\\d+" if you want negatives allowed here
    response.end(...)
} else if("download".equalsIgnoreCase(numberOfLinesString)) {

} else ...
```
```